scriptmodules/supplementary/runcommand/joy2key.py

```diff
@@ -40,7 +40,11 @@ def signal_handler(signum, frame):
 event_format = 'IhBB'
 event_size = struct.calcsize(event_format)
 
-tty_fd = open("/dev/tty", "w")
+try:
+    tty_fd = open('/dev/tty', 'w')
+except:
+    print 'Unable to access /dev/tty, exiting'
+    sys.exit(0)
 js_fd = open(sys.argv[1], "rb")
```

Comment:
Also if following your argument regarding exit codes being valid, then joy2key should not return a 0 if it fails to run, and should return a non 0 error.